services/apps/src/apps/service/apps/de/admin.clj

```diff
@@ -91,8 +91,10 @@
 (defn update-app
   "This service updates high-level details and labels in an App, and can mark or unmark the app as
    deleted or disabled in the database."
-  [{app-id :id :as app}]
+  [{app-name :name app-id :id :as app}]
   (validate-app-existence app-id)
+  (when-not (nil? app-name)
+    (av/validate-app-name app-name app-id (workspace-beta-app-category-id)))
```

Comment:
Yeah, that's a good point. Any exception that occurs inside the transaction macro will cause a rollback, so including the check in the transaction would help. Good catch.